fix: align booking limit timezone between availability and validation

anikdhabal opened the pull request:
## What does this PR do?

Fixes a bug where booking limit availability calculation used a different timezone than booking validation, causing slots to appear available even after reaching the daily booking limit.

devin-ai-integration[bot] committed:
fix: align booking limit timezone between availability and validation

- Use eventType.schedule?.timeZone for booking limits in availability calculation
- Previously used user's timezone causing day boundary mismatch
- Add unit tests to verify timezone alignment
- Fixes issue where slots remain available after reaching booking limit

Co-Authored-By: anik@cal.com <adhabal2002@gmail.com>

anikdhabal committed:
Rename variable limitsTz to eventTimeZone

anikdhabal committed:
update

hariombalhara commented on packages/trpc/server/routers/viewer/slots/util.ts:
```diff
@@ -823,6 +823,8 @@ export class AvailableSlotsService {
     let busyTimesFromLimitsMap: Map<number, EventBusyDetails[]> | undefined = undefined;
     if (eventType && (bookingLimits || durationLimits)) {
       const usersForLimits = usersWithCredentials.map((user) => ({ id: user.id, email: user.email }));
+      const eventTimeZone =
```
<img width="2090" height="234" alt="image" src="https://github.com/user-attachments/assets/80217ad6-81a7-4349-afc4-caca901008ac" />

This is how RegularBookingService computes the timezone, I think we need to be consistent here.

anikdhabal replied:
Addressed, pls check

hariombalhara reviewed (dismissed):
Left a suggestion that I think is blocking

devin-ai-integration[bot] committed:
fix: correct timezone fallback to include user timezone without forcing UTC

- Changed fallback chain to: schedule → event → user → undefined
- Treats empty strings as missing timezones
- Does not force UTC when all timezones are missing
- Aligns with validation behavior when timezone is undefined
- Fixes getSchedule.test.ts failures

Co-Authored-By: anik@cal.com <adhabal2002@gmail.com>

devin-ai-integration[bot] committed:
Revert "fix: correct timezone fallback to include user timezone without forcing UTC"

This reverts commit 721e0bdfd7b08004e68df6bdf5b97b9bd6cf3d5c.

devin-ai-integration[bot] committed:
test: remove complex unit tests with 'as any' casts

- Removed Booking Limits Timezone Alignment tests that required heavy mocking
- These tests relied on spying private methods and had many 'as any' casts
- Timezone alignment is already covered by integration tests in getSchedule.test.ts
- Kept BookingDateInPastError test which has no type issues

Co-Authored-By: anik@cal.com <adhabal2002@gmail.com>

anikdhabal committed:
address review

anikdhabal committed:
Merge branch 'main' into devin/1762057629-fix-booking-limit-timezone-mismatch
